modules/apps/captcha/captcha-taglib/src/main/java/com/liferay/captcha/taglib/servlet/taglib/CaptchaTag.java

```diff
@@ -70,11 +75,32 @@ private String _getURL(HttpServletRequest httpServletRequest) {
 			return _url;
 		}
 
-		ThemeDisplay themeDisplay =
-			(ThemeDisplay)httpServletRequest.getAttribute(
-				WebKeys.THEME_DISPLAY);
+		CaptchaConfiguration captchaConfiguration = null;
+		String url;
 
-		String url = themeDisplay.getPathMain() + "/portal/captcha/get_image";
+		try {
+			captchaConfiguration =
+				ConfigurationProviderUtil.getSystemConfiguration(
+					CaptchaConfiguration.class);
+		}
+		catch (ConfigurationException configurationException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(configurationException);
+			}
+		}
+
+		if ((captchaConfiguration != null) &&
+			captchaConfiguration.enableSimpleCaptchaHeadlessAPI()) {
+
+			url = "/o/captcha/v1.0/simple";
+		}
+		else {
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)httpServletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			url = themeDisplay.getPathMain() + "/portal/captcha/get_image";
```

Comment:
`			ThemeDisplay themeDisplay =
				(ThemeDisplay)httpServletRequest.getAttribute(
					WebKeys.THEME_DISPLAY);

			String url = themeDisplay.getPathMain() + "/portal/captcha/get_image";		
				CaptchaConfiguration captchaConfiguration = null;
`
No need else first init the url

Reply:
Fixed it. Thank